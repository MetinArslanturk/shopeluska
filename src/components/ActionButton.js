import React from 'react';
import { Button } from 'antd';

const ActionButton = ({ type, text, onClickActionFunction, itemToArguments }) => {
    const handleOnclick = () => {
        onClickActionFunction(itemToArguments);
    };
console.log('rendered');
    return (
        <>
            <Button type={type} className="action-button" onClick={handleOnclick}>{text}</Button>
        </>
    );

}

export default React.memo(ActionButton);