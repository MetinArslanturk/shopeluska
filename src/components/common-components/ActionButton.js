import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';

const ActionButton = ({
    type,
    text,
    onClickActionFunction,
    itemToArguments,
}) => {
    const handleOnClick = () => {
        onClickActionFunction(itemToArguments);
    };
    return (
        <>
            <Button
                type={type}
                className="action-button"
                onClick={handleOnClick}
            >
                {text}
            </Button>
        </>
    );
};

ActionButton.propTypes = {
    type: PropTypes.string,
    text: PropTypes.string,
    onClickActionFunction: PropTypes.func,
    itemToArguments: PropTypes.any,
};

export default React.memo(ActionButton);
