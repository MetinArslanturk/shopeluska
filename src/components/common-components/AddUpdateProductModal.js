import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Form, Input, Modal, Select } from 'antd';
import { startAddProduct, startUpdateProduct } from '../../actions/products';

const { Option } = Select;

export const AddUpdateProductModal = ({
    isOpen,
    addProduct,
    updateProduct,
    isUpdate,
    productData,
    handleClose,
    handleSave,
}) => {
    const [form] = Form.useForm();

    const onFinish = (values) => {
        values.price = Number(values.price);
        values.stock = Number(values.stock);

        if (!isUpdate) {
            addProduct(values)
                .then((res) => {
                    handleSave(res);
                })
                .catch(() => {
                    // optional action
                });
        } else {
            values._id = productData._id;
            updateProduct(values)
                .then((res) => {
                    handleSave(res);
                })
                .catch(() => {
                    // optional action
                });
        }
    };

    const initialValues = isUpdate
        ? productData
        : {
              name: '',
              description: '',
              price: 0.0,
              stock: 0,
              imageUrl: '',
              category: undefined,
          };

    // Tricky solution for logical error in ant design setting form initial value dynamically
    if (isOpen) {
        setTimeout(() => {
            form.setFieldsValue(initialValues);
        }, 100);
    }

    return (
        <>
            <Modal
                title={isUpdate ? 'Update Product' : 'Add Product'}
                visible={isOpen}
                onCancel={handleClose}
                okButtonProps={{
                    onClick: () => {
                        form.submit();
                    },
                }}
            >
                <Form name="basic" form={form} onFinish={onFinish}>
                    <Form.Item label="Name" name="name">
                        <Input />
                    </Form.Item>
                    <Form.Item label="Description" name="description">
                        <Input.TextArea rows={4} />
                    </Form.Item>
                    <Form.Item name="category" label="Category" hasFeedback>
                        <Select placeholder="Please select category" removeIcon>
                            <Option value="clothing">Clothing</Option>
                            <Option value="shoes">Shoes</Option>
                            <Option value="accessories">Accessories</Option>
                        </Select>
                    </Form.Item>
                    <Form.Item label="Price" name="price">
                        <Input />
                    </Form.Item>
                    <Form.Item label="Stock" name="stock">
                        <Input />
                    </Form.Item>
                    <Form.Item label="Image Url" name="imageUrl">
                        <Input />
                    </Form.Item>
                </Form>
            </Modal>
        </>
    );
};

AddUpdateProductModal.propTypes = {
    isOpen: PropTypes.bool,
    addProduct: PropTypes.func,
    updateProduct: PropTypes.func,
    isUpdate: PropTypes.bool,
    productData: PropTypes.any,
    handleClose: PropTypes.func,
    handleSave: PropTypes.func,
};

const mapDispatchToProps = (dispatch) => ({
    addProduct: (productData) => dispatch(startAddProduct(productData)),
    updateProduct: (productData) => dispatch(startUpdateProduct(productData)),
});

export default connect(undefined, mapDispatchToProps)(AddUpdateProductModal);
