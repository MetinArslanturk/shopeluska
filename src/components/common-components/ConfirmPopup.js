import { Modal } from 'antd';
const { confirm } = Modal;

export const deleteConfirm = (onOkCallback, onOkArgument) => {
    confirm({
        title: 'Do you want to delete these item?',
        content: 'Items will be removed permanently.',
        onOk: () => {
          onOkCallback(onOkArgument);
        },
        onCancel() {},
      });
}