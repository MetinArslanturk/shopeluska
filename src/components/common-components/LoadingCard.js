import React from 'react';
import { Card, Skeleton } from 'antd';

export const LoadingCard = ({ products }) => (
    <Card>
        <Skeleton active />
        <Skeleton active />
    </Card>
)

export default React.memo(LoadingCard);

