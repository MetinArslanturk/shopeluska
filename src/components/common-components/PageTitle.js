import React from 'react';
import { Helmet } from 'react-helmet';

const defaultTitle = 'Shopeluska';

const PageTitle = React.memo(({ title }) => {
    return (
        <Helmet>
            <title>{title ? title : defaultTitle}</title>
        </Helmet>
    );
});

export { PageTitle };