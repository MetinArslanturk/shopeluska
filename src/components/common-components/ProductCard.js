import React from 'react';
import { connect } from 'react-redux';
import { baseHref } from '../../config/config';
import { Link } from 'react-router-dom';
import { Card, Rate, Button } from 'antd';
import { addToCart } from '../../actions/shopping';


export const ProductCard = ({ product, addToCart }) => {

    const handleAddToCard = (e) => {
        e.stopPropagation();
        e.preventDefault();
        addToCart(product._id);
    }

    return (
        <>
            <Link to={baseHref + 'product/' + product._id}>
                <Card
                    size="small"
                    hoverable
                    cover={<img alt="" className="picture-cover" src={product.imageUrl} />}
                    title={product.name}
                    bordered={true}>
                    <div className="card-description">
                        {product.description.length > 97 ? product.description.slice(0, 97) + '...' : product.description}
                    </div>
                    <div className="card-price">
                        <Rate disabled defaultValue={product.rate} />
                        <div className="card-price-text-wrapper">
                            <span className="card-price-text">{product.price} $</span>
                        </div>
                    </div>
                    <div className="card-add-button">
                        <Button onClick={handleAddToCard}>Add To Card</Button>
                    </div>
                </Card>
            </Link>
        </>
    )
}
const mapDispatchToProps = (dispatch) => ({
    addToCart: (productId) => dispatch(addToCart(productId))
})
export default connect(undefined, mapDispatchToProps)(ProductCard);

