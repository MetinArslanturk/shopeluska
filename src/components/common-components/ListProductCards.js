import React from 'react';
import ProductCard from './ProductCard';
import LoadingCard from './LoadingCard';
import { Row, Col } from 'antd';

const emptyArr = Array.apply(null, Array(12));

export const ListProductCards = ({ products }) => {

    return (
        <>
            <Row gutter={[20, 15]}>
                {products.length === 0 && emptyArr.map((card, indice) => {
                    return (
                        <Col xs={24} sm={24} md={12} lg={12} xl={8} xxl={6} key={indice}>
                            <LoadingCard />
                        </Col>
                    )
                })}
                {products.map((product) => {
                    return (
                        <Col xs={24} sm={24} md={12} lg={12} xl={8} xxl={6} key={product.key}>
                            <ProductCard product={product} />
                        </Col>
                    )
                })}
            </Row>
        </>
    )
}

export default React.memo(ListProductCards);

