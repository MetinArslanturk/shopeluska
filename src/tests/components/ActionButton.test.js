import React from 'react';
import { shallow } from 'enzyme';
import { ActionButton } from '../../components/common-components/ActionButton';

test('should render', () => {
    const wrapper = shallow(<ActionButton type="primary" />);
    expect(wrapper).toMatchSnapshot();
});

test('should trigger function with element', () => {
    const argument = 'test';
    const targetFn = jest.fn();
    const wrapper = shallow(
        <ActionButton
            type="primary"
            onClickActionFunction={targetFn}
            itemToArguments={argument}
        />
    );
    wrapper.find('.action-button').prop('onClick')();
    expect(targetFn).toHaveBeenLastCalledWith(argument);
});
