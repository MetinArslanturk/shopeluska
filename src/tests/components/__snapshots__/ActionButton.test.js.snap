// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<Fragment>
  <Button
    block={false}
    className="action-button"
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  />
</Fragment>
`;
